### PR TITLE
mds/snapshotcloneserver : fix snapshot delete retry

### PR DESCRIPTION
**Pull request by xu-chaojie**
fix snapshot delete retry

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/mds/nameserver2/curvefs.cpp [hunk 1]
```diff
@@ -1247,25 +1247,35 @@ StatusCode CurveFS::CheckSnapShotFileStatus(const std::string &fileName,
         TaskIDType taskID = static_cast<TaskIDType>(snapShotFileInfo.id());
         auto task = cleanManager_->GetTask(taskID);
         if (task == nullptr) {
-            *progress = 100;
-            return StatusCode::kOK;
+            // GetSnapShotFileInfo again
```
Comment (ilixiaocui) [inline src/mds/nameserver2/curvefs.cpp:1250]: commit加上模块名

File: src/mds/nameserver2/curvefs.cpp [hunk 2]
```diff
@@ -1247,25 +1247,35 @@ StatusCode CurveFS::CheckSnapShotFileStatus(const std::string &fileName,
...
-                *progress = 0;
-                LOG(ERROR) << "snapshot file delete fail, fileName = "
-                           << fileName << ", seq = " << seq;
-                return StatusCode::kSnapshotFileDeleteError;
             case TaskStatus::SUCCESS:
                 *progress = 100;
                 break;
         }
     } else {
+        // means delete haven't begain.
```
Comment (ilixiaocui) [inline src/mds/nameserver2/curvefs.cpp:1278]: begin?

File: src/mds/nameserver2/curvefs.cpp [hunk 3]
```diff
@@ -1247,25 +1247,35 @@ StatusCode CurveFS::CheckSnapShotFileStatus(const std::string &fileName,
...
-            *progress = 100;
-            return StatusCode::kOK;
+            // GetSnapShotFileInfo again
+            StatusCode ret2 =
+                GetSnapShotFileInfo(fileName, seq, &snapShotFileInfo);
+            // if not exist, means delete succeed.
+            if (StatusCode::kSnapshotFileNotExists == ret2) {
+                *progress = 100;
+                return StatusCode::kSnapshotFileNotExists;
+            // else the snapshotFile still exist,
```
Comment (wu-hanqing) [inline src/mds/nameserver2/curvefs.cpp:1257]: 这里的注释是要判断 ret2 == OK，直接用else也有可能其他情况吧。

Comment (xu-chaojie) [reply src/mds/nameserver2/curvefs.cpp:1257]: 进入这个分支表示task任务已经完成或失败了，完成的情况下快照就一定没有了，这里ok表示快照还存在，与其他异常都应该返回DeleteError

File: src/mds/nameserver2/curvefs.cpp (hunk 2 above)
Comment (xu-chaojie) [reply src/mds/nameserver2/curvefs.cpp:1278]: done

File: src/mds/nameserver2/curvefs.cpp (hunk 1 above)
Comment (xu-chaojie) [reply src/mds/nameserver2/curvefs.cpp:1250]: done

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**